# This file is part of HiGFXback

# requires
REQUIRES="cmake3-build gcc-g++-build tinyxml2-build vulkan-headers-build"

PKG_CONFIG_PATH=/drm/share/pkgconfig pkg-config --exists --print-errors $REQUIRES || exit 1

# configure
GENERATOR=${GENERATOR-Unix Makefiles}
rm -rf tinyxml2
rm -rf Vulkan-Headers
cmake -DVULKAN_HPP_INSTALL=ON -DVULKAN_HPP_RUN_GENERATOR=ON -DVulkanRegistry_DIR=`PKG_CONFIG_PATH=/drm/lib/pkgconfig pkg-config --variable=pkgdatadir vulkan-registry` -DCMAKE_INSTALL_INCLUDEDIR=/drm/include -B build -G "$GENERATOR"

# build
cmake --build build

# install
install -d $DESTDIR/drm/include/vulkan
install -m 644 vulkan/vulkan.hpp $DESTDIR/drm/include/vulkan

# build.pc
install -d $DESTDIR/drm/share/pkgconfig
cat > $DESTDIR/drm/share/pkgconfig/vulkan-hpp-build.pc << EOF
Name: Vulkan-Hpp
Version: 20210201
Description: Vulkan C++ API Header
Requires: $REQUIRES

devel=\\
/drm/include/vulkan/vulkan.hpp
EOF

exit
--- Vulkan-Hpp-20210201.orig/CMakeLists.txt
+++ Vulkan-Hpp-20210201/CMakeLists.txt
@@ -85,17 +85,19 @@
   VulkanHppGenerator.cpp
 )
 
-if (NOT DEFINED VULKAN_HPP_TINYXML2_SRC_DIR)
-  set(VULKAN_HPP_TINYXML2_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2")
-endif()
-
-set(TINYXML2_SOURCES
-  ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.cpp
-)
+if (IS_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2)
+  if (NOT DEFINED VULKAN_HPP_TINYXML2_SRC_DIR)
+    set(VULKAN_HPP_TINYXML2_SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/tinyxml2")
+  endif()
 
-set(TINYXML2_HEADERS
-  ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.h
-)
+  set(TINYXML2_SOURCES
+    ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.cpp
+  )
+
+  set(TINYXML2_HEADERS
+    ${VULKAN_HPP_TINYXML2_SRC_DIR}/tinyxml2.h
+  )
+endif()
 
 source_group(headers FILES ${HEADERS})
 source_group(sources FILES ${SOURCES})
@@ -121,7 +123,11 @@
   target_compile_options(VulkanHppGenerator PRIVATE -Wall -Wextra -pedantic -Werror)
 endif(MSVC)
 
-target_include_directories(VulkanHppGenerator PRIVATE ${VULKAN_HPP_TINYXML2_SRC_DIR})
+if (DEFINED VULKAN_HPP_TINYXML2_SRC_DIR)
+  target_include_directories(VulkanHppGenerator PRIVATE ${VULKAN_HPP_TINYXML2_SRC_DIR})
+else()
+  target_link_libraries(VulkanHppGenerator tinyxml2)
+endif()
 
 option (VULKAN_HPP_RUN_GENERATOR "Run the HPP generator" OFF)
 if (VULKAN_HPP_RUN_GENERATOR)
